consensus/src/round_manager_test.rs

```diff
@@ -1022,3 +1022,48 @@ fn safety_rules_crash() {
         node.next_proposal().await;
     });
 }
+
+#[test]
+fn echo_timeout() {
+    let mut runtime = consensus_runtime();
+    let mut playground = NetworkPlayground::new(runtime.handle().clone());
+    let mut nodes = NodeSetup::create_nodes(&mut playground, runtime.handle().clone(), 4);
+    runtime.spawn(playground.start());
+    timed_block_on(&mut runtime, async {
+        // clear the message queue
+        for node in &mut nodes {
+            node.next_proposal().await;
+        }
+        // timeout 3 nodes
+        for node in &mut nodes[1..] {
+            node.round_manager
+                .process_local_timeout(1)
+                .await
+                .unwrap_err();
+        }
+        let node_0 = &mut nodes[0];
+        // node 1 doesn't timeout and should echo the timeout after 2 timeout message
+        for i in 0..2 {
```

Comment:
should we check that it doesn't send the echo timeout again after receiving a 3rd timeout?